### DISCUSSION TITLE
Limit one kind of a job at the time

### DISCUSSION THREAD

**Discussion by zMcKracken** [jobrunr/jobrunr#1120] [Q&A]
Hello everyboy, 
I have a particular situation and I'm wondering how or if I can solve it using jobrunr. I'm not really into queue system so my apologize for saying stupid things.
I have 2 jobs that are submitted one after the other at very close time, but If they belong, let's say, to the same type and id only one can be submitted while the other must wait until the first is done.
Before submitting the second I can filter the storage provider for state ENQUEUED and PROCESSING and if I found one job with the same type and id I can delay and keep delaying the submission until the first is done. But I don't think this is the best approach and moreover it does not guarantee what I'm looking for.

**Comment by rdehuyss** [comment]
That's the mutex functionality of [JobRunr Pro](https://www.jobrunr.io/en/documentation/pro/mutexes/)...
